### ISSUE TITLE
Intro to TFLearn - 2. ReLU and Softmax typo

### ISSUE THREAD

**Issue by nehal96** [udacity/dlnd-issue-reports#114]
In the Activation Functions section, you'll begin by explaining the sigmoid function and its derivative, and its drawbacks:

> As noted in the backpropagation material, the derivative of the sigmoid maxes out at 0.25 (see above). This means when you're performing backpropagation with sigmoid units, the errors going back into the network will be shrunk by at least a quarter at every layer.

Shouldn't it be *at most* a quarter? That is, when the error is highest (sigmoid function returns 0.5, error is 0.5 or -0.5), the derivative is at its maximum of 0.25, and hence it *at most* reduces by a quarter (and so forth..).

**Comment by tangrui** [comment]
Agreed, I think it should be at most also.

**Comment by clapollo** [comment]
This isn't wrong, but we'll need to come up with some math to explain why. First, it's not talking about the values it reduces, it's talking about the magnitude of the gradient that gets passed back through the layers. (Also, if it *were* talking about the values, 0.25 would reduce by 75%, and a value smaller than 0.25 would reduce by even more.)

Again, this needs some explanation. Will try to get that sorted.

**Comment by nehal96** [comment]
So the magnitude of the gradient would at least be 0.25? Wouldn't that still not be the minimum? 

Sorry, I may have got this all confused up. Yes, a clearer explanation with some math would be much appreciated. Cheers!

**Comment by clapollo** [comment]
Hehe, not the magnitude of the gradient goes *down* by at least 25% . It is super confusing. I hope to improve it soon (but I honestly need to figure out how to explain it myself, or find someone who can do so clearly)  :D

**Comment by onemodtwo** [comment]
Because of the chain rule, the gradient of ANY activation gets multiplied into the backpropagated error as a scalar. For sigmoid(x), the value of the gradient varies between 0 as x --> +/- infinity and 1/4 at x = 0. Therefore, the MINIMUM diminution of the backpropagated error occurs at the MAXIMUM value of the gradient because 0.25*error is greater than g*error for any g in (0, 0.25).

**Comment by clapollo** [comment]
We've updated the text a bit. I think we also had some incorrect wording, because multiplying something by .25 reduces it by 75%, not 25%. Using a smaller number, like .1, reduces the value by 90%. So the correct reduction should be at least 75% (three quarters) at each level.

The real point here is that, if you try using a few sigmoid layers in a row, your gradient will basically disappear. :)
